src/backend/functions/orchestrator/lib/index.js

```diff
@@ -102,35 +98,103 @@ function createDeleteParams({mediaUrl, tasksTableName}) {
+      },
+      ReturnValues: 'ALL_NEW'
+   };
+
+   return updateItem(params)
+       .then(({Attributes}) => ({...Attributes, tasksTableName}))
+});
 
+const unrecoverableError = R.curry((updateItem, {tasksTableName, MediaUrl}) => {
+   const params = {
+      TableName: tasksTableName,
+      Key: {
+         MediaUrl: MediaUrl
+      },
+      UpdateExpression: 'SET TaskStatus = :status',
+      ExpressionAttributeValues: {
+         ':status':  'UNRECOVERABLE_ERROR'
+      },
+      ReturnValues: 'ALL_NEW'
+   };
+
+   return updateItem(params);
+});
+
+// error :: AWS.ECS -> AWS.DynamoDB -> {k: v} -> {k: v} -> Promise {k:v}
+const error = R.curry((ecs, updateItem) =>
+    handle(composeP(errorUpdate(updateItem), startTranscription(ecs)), ({MediaUrl: mediaUrl}) => ({mediaUrl}))
+);
 
 // handler :: AWS.ECS -> AWS.DynamoDB -> {k: v} -> ({k: v} -> {k: v} -> Promise {k: v})
 module.exports = (ecs, ddb, env) => {
+   const update = params => ddb.update(params).promise();
+
    const handlers = {
-      WAITING: waiting(ecs, params => ddb.update(params).promise()),
+      WAITING: waiting(ecs, update),
       INITIALIZING: noop,
       PROCESSING: noop,
       TERMINATING: terminating(ecs),
       TERMINATED: terminated(params => ddb.delete(params).promise()),
-      ERROR: error
+      ERROR: error(ecs, update),
+      UNRECOVERABLE_ERROR: noop
    };
 
-   return (event, context) => {
+   return event => {
+
       const promises = R.map(record => {
          const image = converter.unmarshall(R.path(['dynamodb', 'NewImage'], record));
-         const handler = R.defaultTo(noop, handlers[image.TaskStatus]);
-         return handler(env, image);
+         const {MediaUrl, Retries, TaskStatus} = image;
+         const {TASKS_TABLE_NAME: tasksTableName} = env;
+
+         console.log('Status: ' + TaskStatus);
+
+         if(R.defaultTo(0, Retries) > parseInt(env.RETRY_THRESHOLD)) {
+            console.log('Error threshold exceeded');
+            return unrecoverableError(update, {MediaUrl, tasksTableName});
+         }
+
+         const handler = R.defaultTo(noop, handlers[TaskStatus]);
+         return handler(env, image).catch(err => {
+            const e = new Error(err.message);
+            e.mediaUrl = MediaUrl;
+            e.tasksTableName = tasksTableName;
+            return e;
+         });
       }, event.Records);
 
       return Promise.all(promises)
+          .then(R.filter(x => x.name === 'Error'))
+          .then(R.forEach(err => update({
+             TableName: err.tasksTableName,
+             Key: {
+                MediaUrl: err.mediaUrl
+             },
+             UpdateExpression: 'SET TaskStatus = :status',
+             ExpressionAttributeValues: {
+                ':status':  'ERROR'
+             },
+             ReturnValues: 'ALL_NEW'
+          })))
           // this is temporary but important to remember that any uncaught DynamoDb stream errors block
           // the queue for 24 hrs
```

Comment:
I didn't know this!! good to know. is this also the case if you configure a DLQ on the function?